### PR TITLE
Variants for startup options

### LINKED ISSUE #892: Create variants for startup options
Please see the latest version of J3Cub and the way I did the QT launcher variants. You can now start the aircraft in a predefined configuration and still have the flexibility to change the configuration on the fly.
If we choose to do this I would suggest we come up with a list of the best, smallest list to showcase the available configurations and limit the amount of variants.

### LINKED ISSUE #967: Recreate splash screens
Given that we have made considerable improvements to the external 3D model since the splash screens have been created, I propose we re-create them in order to showcase how our aircraft is currently looking. 

@wlbragg wrote:

> I should do some variant startup states, that would go hand in hand with the previews and splash screen work.
> 
> The variants I would include are
> 
>     Default (160 engine, default tires)
>     Bush (180 engine, 26" tires)
>     Float (160 engine, floats)
>     Amphibian (180 engine, amphibious floats)
>     Snow Skis (160 or 180 engine, skis)
> 
> I would have a different splash screen for each.
> I would have at least one preview for each of the above and also include a preview for the 36" tire.

### PR DESCRIPTION
**Pull request by wlbragg**
Fixes #892 
Fixes #967

About all that is needed here is some new splash and preview images that fit the variant theme.

### CODE REVIEW

**Comment by gilbertohasnofb** [comment]
> About all that is needed here is some new splash and preview images that fit the variant theme.

I will happily work on that but I think we should do that only after the internal uplifting since that might be visible from the outside. Is that ok with you?

File: c172p-amphibious-set.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+c172p simulation configuration. This files ties together all the components
+used by FGFS to represent the c172p. Components include the flight data
+model, instrument panel, and external 3D model.
+************************************************************************
+-->
+
+<PropertyList include="c172p-set.xml">
+    <sim>
+  	    <variant-of>c172p</variant-of>
+        <description>Cessna 172p (180hp) Amphibious (JSBSim)</description>
```
[the remark this answers is not in the document]
Comment (wlbragg) [reply c172p-amphibious-set.xml]: >I can fix this if you're ok with it.

That fine with me!

**Comment by onox** [comment]
Some problems:

- FG still selects random previews
- `/sim/model/variant` isn't modified when switching in "Aircraft Options". I tried to set the value in `set_bushkit()` in `Nasal/physics.nas` but that didn't work.

**Comment by wlbragg** [comment]
@gilbertohasnofb I'm waiting on this until after we create any new previews. At that time I'll also review and address @onox concerns.

**Comment by gilbertohasnofb** [comment]
@wlbragg if you want you can start working on this again. I can push the new previews and splash screens to this very branch here. I should get it done in the next few days (sorry for the delay, I've spent ages testing PRs and fixing those tutorials yesterday). Would that work for you? This way we can start addressing @onox's comments while I work on the images.

**Comment by wlbragg** [comment]
@gilbertohasnofb I have already started on this. I'm not getting the same results @onox posted about. There was still a bit of work to do on this and after I am done and push the modifications you or @onox can see if those concerns were addressed.
So to answer your question, yes that works for me.

**Comment by wlbragg** [comment]
@gilbertohasnofb @onox 
On my system this appears to be working correctly. I see no misbehaving as reported by @onox.

>FG still selects random previews

If you mean what I think you do then this is now fixed but the images won't fall in line with the variant until @gilbertohasnofb creates the images for me to link to.
But to test if it is working correctly, I know the splash for each variant are set up as follows.

Default /26"/36"  splash image will be a default tired image.
Float/Amphibious image will be the same float image.
Ski image will be a ski image.

All the variants should have the same 7 or 8 preview to cycle through using the arrows.

**Comment by gilbertohasnofb** [comment]
@wlbragg I will create the splash screens and previous as soon as possible. Today will be a hectic day but I should have it all done by tomorrow. I will push them to this PR here.

**Comment by gilbertohasnofb** [comment]
@wlbragg the new splash screens have been pushed and I also updated the default `-set.xml` file. But for some reason the sim was still showing me the old previews, even though those files are not in this branch any longer... could you help me investigate what's going on? Also, you may need to add the splash screens to the other -set files. Personally I would have a specific preview for each variant in the launcher but a random splash from all 13 options when launching the sim, regardless of the variant.

**Comment by wlbragg** [comment]
@gilbertohasnofb
I think this is ready to go.

**Comment by gilbertohasnofb** [comment]
@wlbragg This PR is marked in yellow with the label `Changes requested`, are you sure this is ready to merge?

**Comment by wlbragg** [comment]
@gilbertohasnofb as far as I know, yes, it is done. @onox made some of the changes he requested himself. Other observations he made I don't or can't reproduce. As far as I can tell this is finished and behaving as expected.
I guess I would feel more comfortable at this point if @onox signed off on this only because he flagged it, but it is pretty straight forward stuff.

**Comment by onox** [comment]
- FG still selects random previews (preview of regular variant when you start float/amphibious for example)
~~- `/sim/model/variant` isn't modified when switching in "Aircraft Options" (the user doesn't see that the model changes). I tried to set the value in `set_bushkit()` in `Nasal/physics.nas` but that didn't work.~~
- Bush26 and Bush36 are ok, but the ~~floats~~, amphibious and ~~skis~~ are sunk through the runway ~~and in the dialog Default + 160 HP is selected.~~

About the amphibious variant: 80 % of the front wheels are below the runway and 60 % of the rear wheels. When I try to take off, I go 20 to 30 kts for a long time, eventually reaching 50 to 60 kts. I see yellow spark in the external view.

I don't use the Qt launcher, just `fgfs`.

**Comment by wlbragg** [comment]
>FG still selects random previews (preview of regular variant when you start float/amphibious for example)

@gilbertohasnofb indicated he preferred to see a random splash screen. However when you choose the variant it does show you the correct image.

>About the amphibious variant: 80 % of the front wheels are below the runway and 60 % of the rear wheels. When I try to take off, I go 20 to 30 kts for a long time, eventually reaching 50 to 60 kts. I see yellow spark in the external view.

Is this only in this branch that you see this? Can anyone else reproduce, because I don't see it?

**Comment by gilbertohasnofb** [comment]
> @gilbertohasnofb indicated he preferred to see a random splash screen. However when you choose the variant it does show you the correct image.

Splash screens should be randomly selected out of the 13 options, else what's the purpose of having them at all? My suggestion is to use a specific image for each variant on the preview only (i.e. in the launcher) but use a random splash when launching the sim. @wlbragg has a list of which image should be associated with each preview in the launcher.

**Comment by wlbragg** [comment]
>@wlbragg has a list of which image should be associated with each preview in the launcher.

If you pull this branch it is all setup, each variant will show it's custom image when you select that variant, but if you scroll through the previews you will see all the images. Then when you click on the "Fly" button you will see a random splash screen.

There are two options for the splash while it is loading and I could go either way.

One, the way @gilbertohasnofb suggested is to have it show random images.
>Splash screens should be randomly selected out of the 13 options, else what's the purpose of having them at all?
The other reason to have them all is for preview while in the launcher, you can scroll through them there to get an idea as to what the aircraft has to offer.

Two, you can have only one image (or a couple predetermined images, "random") show in the splash screen related to that variant.

Like I said, I am OK with either option so I guess it's up to the two of you to decide at this point how we should do it.

>About the amphibious variant: 80 % of the front wheels are below the runway and 60 % of the rear wheels. When I try to take off, I go 20 to 30 kts for a long time, eventually reaching 50 to 60 kts. I see yellow spark in the external view.

@onox, if this only happens in this branch It would be easiest for me to start over with master and add the variant files and code. I honestly have no idea why you would have this kind of an issue because of the variant work.

**Comment by gilbertohasnofb** [comment]
> If you pull this branch it is all setup, each variant will show it's custom image when you select that variant, but if you scroll through the previews you will see all the images. Then when you click on the "Fly" button you will see a random splash screen.

That sounds perfect.

> There are two options for the splash while it is loading and I could go either way.

My vote still goes for the first option, but it's up to you all implementing this.

**Comment by onox** [comment]
If there are multiple people that don't have the problem that I have with the amphibious variant, then it might be a problem on my end. Either I need to recompile FG or delete the settings file in `.fgfs`. But I don't have much time for FG these days. So you can go ahead and merge it if there are no other problems.

**Comment by wlbragg** [comment]
@onox thanks, I'm pretty sure it is OK. I suspect your local version needs a good house cleaning!

**Comment by gilbertohasnofb** [comment]
@onox I tried again but I can't reproduce your problem with the amphibious.

@wlbragg This looks good to be merged but there is one thing I would change: I would use `c172p-preview2.jpg` instead of `c172p-preview0.jpg` for the default variant. Would that be all right with you?

**Comment by wlbragg** [comment]
>I would use c172p-preview2.jpg instead of c172p-preview0.jpg for the default variant. Would that be all right with you?

Yes, I'll change it.

**Comment by gilbertohasnofb** [comment]
@wlbragg Thanks! Once that's done do let me know and I can merge this

**Comment by gilbertohasnofb** [comment]
Also, can we dismiss this review by @onox or do you still have something to fix in relation to it?

**Comment by gilbertohasnofb** [comment]
@wlbragg Great work, thanks a lot! 

> As far as I know the only thing that we still might be able to get into this release is the state stuff. 

That would be excellent. I think the aircraft is looking really good and stable, this will be a solid release!

**Pull request merged**